### PR TITLE
AB#3370 Add confirmation code backend project

### PR DESCRIPTION
**Pull request by ESDC-Olivier-Roy**
### Description

Add the confirmation code on the backend project. This include repository, service and controller class with a single endpoint for the moment.

### Related Azure Boards Work Items
[AB#3370](https://dev.azure.com/DTS-STN/1fc40a8f-28cf-47bc-b6e4-1c234bd06177/_workitems/edit/3370) ### Screenshots (if applicable)

### Checklist

- [x] I have tested the changes locally
- [x] I have updated the documentation if necessary
- [x] I have added/updated tests that prove my fix is effective or that my feature works
- [ ] I have checked that my code follows the project's coding style by running `npm run format:check`
- [ ] I have checked that my code contains no linting errors by running `npm run lint`
- [ ] I have checked that my code contains no type errors by running `npm run typecheck`
- [ ] I have checked that all unit tests pass by running `npm run test:unit -- run`
- [ ] I have checked that all e2e tests pass by running `npm run test:e2e`

### CODE REVIEW

File: backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/ConfirmationCodeModel.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
+package ca.gov.dtsstn.cdcp.api.web.v1.model;
+import java.time.Instant;
+
+import org.immutables.value.Value.Immutable;
```
Comment (CanadianPhilipBarkow) [inline backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/ConfirmationCodeModel.java:4]: Unused import here.

File: backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/ConfirmationCodeModelMapper.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package ca.gov.dtsstn.cdcp.api.web.v1.model.mapper;
+import java.util.List;
+import java.util.stream.StreamSupport;
```
Comment (CanadianPhilipBarkow) [inline backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/ConfirmationCodeModelMapper.java:3]: And unused imports here.

**Review by CanadianPhilipBarkow — CHANGES_REQUESTED** [review]
Needs a rebasing.

File: backend/src/main/java/ca/gov/dtsstn/cdcp/api/data/entity/ConfirmationCodeEntity.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
+package ca.gov.dtsstn.cdcp.api.data.entity;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.ManyToOne;
+import java.util.Date;
+import java.time.Instant;
+import java.time.LocalDateTime;
+
+import org.immutables.builder.Builder;
+import org.springframework.cglib.core.Local;
```
[the remark this answers is not in the document]
Comment (ESDC-Olivier-Roy) [reply backend/src/main/java/ca/gov/dtsstn/cdcp/api/data/entity/ConfirmationCodeEntity.java]: imports removed

File: backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/ConfirmationCodeModel.java (hunk 1 above)
Comment (ESDC-Olivier-Roy) [reply backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/ConfirmationCodeModel.java]: imports removed

File: backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/ConfirmationCodeModelMapper.java (hunk 2 above)
Comment (ESDC-Olivier-Roy) [reply backend/src/main/java/ca/gov/dtsstn/cdcp/api/web/v1/model/mapper/ConfirmationCodeModelMapper.java]: imports removed

**Review by CanadianPhilipBarkow — APPROVED** [review]

**Pull request merged**